### DISCUSSION TITLE
Using multiple JSONSchema types in a single file

### DISCUSSION THREAD

**Discussion by alignsoft** [cyrilletuzi/angular-async-local-storage#62] [Q&A] [answered]
In my Service files, I have multiple different data types that I'm fetching from localstorage via the `getItem<type>(key, { schema })` syntax.

I've defined the following:

```
const schema: JSONSchemaString = { type: 'string' };
const schemaNum: JSONSchemaNumeric = { type: 'number' };
const schemaObj: JSONSchemaObject = {
    properties: {
      crypt: { type: 'string' } as JSONSchemaString,
    },
    required: ['crypt']
  };
```

And I get the following error when I try to use anything but 'schema' with getItem, for example:
`getItem<number>(keyName, { schemaNum })`

```
Argument of type '{ schemaNum: JSONSchemaNumeric; }' is not assignable to parameter of type 'LSGetItemOptions'.  Object literal may only specify known properties, and 'schemaNum' does not exist in type 'LSGetItemOptions'. [2345]
(property) schemaNum: JSONSchemaNumeric
```
Just using `getItem<number>(keyName, { schema })` works fine, but I have multiple different validation signatures that I need to use.

Am I going about this the wrong way, or misunderstanding how this works?

**Comment by cyrilletuzi** [comment, marked as answer]
In ES6+, this:
```typescript
getItem<number>(keyName, { schema })
```

is a shortcut for this:
```typescript
getItem<number>(keyName, { schema: schema })
```

which works only if the property and the variable have the same name. So if your variable has another name, you can't use the shortcut, you need to do this:
```typescript
getItem<number>(keyName, { schema: schemaNum })
```

Added the info in doc.
